File: PREGUNTA3/fsuperior.py
def fibo(n):
    a = 0
    b = 1
    c = 1
    for i in range(n):
        print(a, end=" ")
        a = b
        b = c
        c = a+c 

def fibo_rec(n):
    if n==0:
        return 0
    elif n==1:
        return 1
    elif n==2:
        return 1
    else:
        return fibo_rec(n-1)+fibo_rec(n-2)

def generar(n):
    for i in range(0,n+1):
        print(fibo_rec(i), end=" ")

def fsuperior(funcion1, n):
    funcion1(n)

n = int(input("Introduzca un numero"))
fsuperior(fibo, 5)
fsuperior(generar, 5)

File: PREGUNTA3/fsuperior.scala
object Hola {

  def fsuperior(funciontipo:(Int)=>Unit):Unit={
    funciontipo(5)
  }

  def fiboe(x:Int):Unit={
    var a = 0
    var b = 1
    var c = 1
    for(w <- 0 to x){
      print(a+" ")
      a=b
      b=c
      c=a+b   
    }
  }
  def fibo(x:Int):Int={
    if(x==0) return 0 
    else if(x==1) 1
    else if(x==2) 1
    else fibo(x-1) + fibo(x-2)
  }

  def generar(x:Int):Unit={
    for(w <- 0 to x){
      print(fibo(w)+" ")    
    }
  }
  
  def main(args:Array[String]):Unit = {
    println("Hola mundo")
    println("Dame un valor a: ")
    val a = scala.io.StdIn.readInt()
    fsuperior(fiboe)
    println()
    fsuperior(generar)
  }
  
}